### PR TITLE
Enhance SSL_client_hello_get0_ext to support all extensions

### PR DESCRIPTION
**Pull request by mshekiwy**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: ssl/ssl_lib.c [hunk 1]
```diff
@@ -6677,25 +6677,32 @@ int SSL_client_hello_get_extension_order(SSL *s, uint16_t *exts, size_t *num_ext
 int SSL_client_hello_get0_ext(SSL *s, unsigned int type, const unsigned char **out,
                        size_t *outlen)
 {
-    size_t i;
-    RAW_EXTENSION *r;
+    PACKET pkt,ext_data;
```
Comment (paulidale) [inline ssl/ssl_lib.c:6680]: style nit: space after the comma.

Comment (mshekiwy) [reply ssl/ssl_lib.c]: done

**Comment by mshekiwy** [comment]
any update?

**Comment by nhorman** [comment]
can you rebase so that you can pick up the changes addressing the provider breakage and pass ci?

**Comment by mshekiwy** [comment]
done

**Comment by t8m** [comment]
Please avoid merge commits. Instead use `git rebase master; git  push --force`.

**Pull request closed without merging**